tests/test_regression.py

```diff
@@ -0,0 +1,197 @@
+
+
+@pytest.fixture
+def non_array_input():
+    return [aConstant, aString, aDict, aSet, aBool]
+
+
+# test with deafault parameters
+def test_output_shape_type():
+    scores = get_label_quality_scores(labels=labels, predictions=predictions)
+    assert labels.shape == scores.shape
+    assert isinstance(scores, np.ndarray)
+
+
+def test_labels_are_arraylike(non_array_input):
+    for new_input in non_array_input:
+        with pytest.raises(ValueError) as error:
+            get_label_quality_scores(labels=new_input, predictions=predictions)
+            assert error.type == ValueError
+
+
+def test_predictionns_are_arraylike(non_array_input):
+    for new_input in non_array_input:
+        with pytest.raises(ValueError) as error:
+            get_label_quality_scores(labels=labels, predictions=new_input)
+            assert error.type == ValueError
+
+
+# test for input shapes
+def test_input_shape_labels():
+    with pytest.raises(AssertionError) as error:
+        get_label_quality_scores(labels=labels[:-1], predictions=predictions)
+    assert (
+        str(error.value)
+        == f"Number of examples in labels {labels[:-1].shape} and predictions {predictions.shape} are not same."
+    )
+
+
+def test_input_shape_predictions():
+    with pytest.raises(AssertionError) as error:
+        get_label_quality_scores(labels=labels, predictions=predictions[:-1])
+    assert (
+        str(error.value)
+        == f"Number of examples in labels {labels.shape} and predictions {predictions[:-1].shape} are not same."
+    )
+
+
+# test individual scoring functions
+@pytest.mark.parametrize(
+    "scoring_funcs",
+    [get_residual_score_for_each_label, get_outre_score_for_each_label],
+)
+def test_individual_scoring_functions(scoring_funcs):
+    scores = scoring_funcs(labels=labels, predictions=predictions)
+    assert labels.shape == scores.shape
+    assert isinstance(scores, np.ndarray)
+
+
+# test for method argument
+@pytest.mark.parametrize(
+    "method",
+    [
+        "residual",
+        "outre",
+    ],
+)
+def test_method_pass_get_label_quality_scores(method):
+    scores = get_label_quality_scores(labels=labels, predictions=predictions, method=method)
+    assert labels.shape == scores.shape
+    assert isinstance(scores, np.ndarray)
+
+
+@pytest.mark.parametrize(
+    "method",
+    [
+        "residual",
+        "outre",
+    ],
+)
+def test_expected_scores(method):
+    # characterization test
+    scores = get_label_quality_scores(
+        labels=small_labels, predictions=small_predictions, method=method
+    )
+    assert np.allclose(scores, expected_scores[method], atol=1e-08)
+
+
+def test_cleanlearning():
+    # test fit and predict
+    cl = CleanLearning()
+    cl.fit(X, y)
+    predictions = cl.predict(X)
+    cl_r2_score = cl.score(X, y)
+    manual_r2_score = r2_score(predictions, y)
+    assert len(predictions) == len(y)
+    assert isinstance(cl_r2_score, float)
+    assert cl_r2_score == manual_r2_score
+
```

Comment:
somewhere we need to test the label-error-detection accuracy of this right?  Maybe there is already such a test somewhere but I didn't see such an assert that actually checks whether the performance is good or bad on a toy dataset with obvious errors.

Reply:
Added!